src/elfpy/markets.py

```diff
@@ -181,50 +115,57 @@ def trade_and_update(self, agent_action: MarketAction, pricing_model: PricingMod
         self.check_action_type(agent_action.action_type, pricing_model.model_name())
         # TODO: check the desired amount is feasible, otherwise return descriptive error
         # update market variables which may have changed since the user action was created
-        time_remaining = time_utils.get_yearfrac_remaining(self.time, agent_action.mint_time, self.token_duration)
-        stretched_time_remaining = time_utils.stretch_time(time_remaining, self.time_stretch_constant)
+        time_remaining = StretchedTime(
+            # TODO: We should improve the StretchedTime API so that it accepts yearfracs in
```

Comment:
@sentilesdal has been advocating for moving away from yearfracs in general, and just reporting seconds. With the StretchedTime class we can probably include nice unit converters so people can use whatever units they like? Worth thinking about if we're already addressing the API.

Reply:
We can address these changes in follow-up PRs (as discussed offline).